bluetooth/drivers/bluetoothctl.c

```diff
@@ -0,0 +1,214 @@
+/*  RetroArch - A frontend for libretro.
+ *
+ *  RetroArch is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  RetroArch is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with RetroArch.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <compat/strl.h>
+#include <configuration.h>
+
+#include "../bluetooth_driver.h"
+#include "../../retroarch.h"
+
+static bool bluetoothctl_cache[256]       = {0};
+static unsigned bluetoothctl_counter[256] = {0};
+static struct string_list* lines          = NULL;
+static char command[256]                  = {0};
+
+static void *bluetoothctl_init(void)
+{
+   return (void*)-1;
+}
+
+static void bluetoothctl_free(void *data)
+{
+   (void)data;
+}
+
+static bool bluetoothctl_start(void *data)
+{
+   (void)data;
+   return true;
+}
+
+static void bluetoothctl_stop(void *data)
+{
+   (void)data;
+}
+
+static void bluetoothctl_scan(void)
+{
+   char line[512];
+   union string_list_elem_attr attr;
+   FILE *dev_file                   = NULL;
+
+   attr.i = 0;
+   if (lines)
+      free(lines);
+   lines = string_list_new();
+
+   pclose(popen("bluetoothctl -- power on", "r"));
```

Comment:
Ah, yes... Using the shell may be a sub-optimal solution here. Using a lib for it would be best. I'll be able to test this in the mean time later on today.

Reply:
I'm not sure libbluetooth should be used at all since it's long been deprecated in favor of the dbus api. This particular driver should only be used in case libdbus is not available. The bluez driver should be used 99% of the time.

Reply:
I can just remove this driver since popen() is not considered acceptable, and only leave the dbus one.

Reply:
Alright, how about this then - for now we'll go with this. If in the future there are better solutions, we can always update this later on. Everybody can agree with that?

Reply:
We are using a similar solution with `xrandr`, so it could be a working solution until we swap it out later on :+1: